Source/JavaScriptCore/wasm/WasmBBQJIT64.cpp

```diff
@@ -1630,6 +1630,7 @@ void BBQJIT::emitAllocateGCArrayUninitialized(GPRReg resultGPR, uint32_t typeInd
         ASSERT(hasOneBitSet(elementSize));
         m_jit.lshift64(sizeLocation.asGPR(), TrustedImm32(getLSBSet(elementSize)), scratchGPR);
         m_jit.add64(TrustedImm64(sizeof(JSWebAssemblyArray)), scratchGPR);
+        slowPath.append(m_jit.branch64(CCallHelpers::Above, scratchGPR, CCallHelpers::TrustedImm64(Wasm::maxArraySizeInBytes)));
```

Comment:
Do we need to consider the size of JSWebAssemblyArray also? Since previously we check the element size only
```
if (UNLIKELY(productOverflows<uint32_t>(elementSize, size) || elementSize * size > Wasm::maxArraySizeInBytes))
            return std::nullopt;
```

Reply:
Oh! Nice catch, yeah, we should check with array size.